activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/ProcessEngineWrapper.java

```diff
@@ -200,6 +200,26 @@ public void setTaskVariablesLocal(SetTaskVariablesCmd setTaskVariablesCmd) {
                                       setTaskVariablesCmd.getVariables());
     }
 
+    /**
+     * Delete task by id.
+     * @param taskId the task id to delete
+     */
+    public void deleteTask(String taskId) {
+        Task task = getTaskById(taskId);
+        if (task == null) {
+            throw new ActivitiObjectNotFoundException("Unable to find task for the given id: " + taskId);
+        }
+
+        checkWritePermissionsOnTask(task);
+
+        taskService.deleteTask(taskId,
+                               "Cancelled by " + authenticationWrapper.getAuthenticatedUserId());
+    }
+
+    public void checkWritePermissionsOnTask(Task task) {
+        //TODO: to check the user write permissions on task
```

Comment:
@constantin-ciobotaru here we need to check if the one cancelling the task is the assigned user or an admin. Now everyone can delete the task right?